src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -237,10 +264,6 @@ def _check_preconditions(self, restart_on_demand):
 
             return True
 
-        if latest_task['source_data_dir'] != self.source_data_dir:
```

Comment:
This seems to be irrelevant at this point. If the previous job failed/canceled/timeout, it's okay to restart the job with a different argument. If the previous job is completed/running/pending, the job won't run unless `force`. If the user force runs the job, it also becomes irrelevant.

Reply:
Hmm, what about the case where the job failed on 1 genome but all the others are ok?

Reply:
we do have a ticket for that situation. But I haven't thought through how we can do this. 
https://kbase-jira.atlassian.net/browse/RE2022-96
I feel this still has nothing to do with `source_data_dir`.

Reply:
added back `source_data_dir` check.

Reply:
:+1: